src/attribution/onINP.ts

```diff
@@ -233,6 +235,100 @@ const getIntersectingLoAFs = (
   return intersectingLoAFs;
 };
 
+const getLoAFSummary = (attribution: INPAttribution) => {
+  const loafAttribution: LongAnimationFrameSummary = {};
+
+  // Stats across all LoAF entries and scripts.
+  const interactionTime = attribution.interactionTime;
+  const inputDelay = attribution.inputDelay;
+  const processingDuration = attribution.processingDuration;
+  let totalNonForcedStyleAndLayoutDuration = 0;
+  let totalForcedStyleAndLayout = 0;
+  let totalScriptTime = 0;
+  let numScripts = 0;
+  let slowestScriptDuration = 0;
+  let slowestScriptEntry!: PerformanceScriptTiming;
+  let slowestScriptPhase: string = '';
+  const phases = {} as Record<string, Record<ScriptInvokerType, number>>;
+
+  attribution.longAnimationFrameEntries.forEach((loafEntry) => {
+    totalNonForcedStyleAndLayoutDuration +=
+      loafEntry.startTime + loafEntry.duration - loafEntry.styleAndLayoutStart;
+    loafEntry.scripts.forEach((script) => {
+      const scriptEndTime = script.startTime + script.duration;
+      if (scriptEndTime < interactionTime) {
+        return;
+      }
+      const intersectingScriptDuration =
+        scriptEndTime - Math.max(interactionTime, script.startTime);
+      totalScriptTime += intersectingScriptDuration;
+      numScripts++;
+      totalForcedStyleAndLayout += script.forcedStyleAndLayoutDuration;
+      const invokerType = script.invokerType;
+      let phase = 'processingDuration';
```

Comment:
worth typing `phase` to avoid the cast below

Reply:
Done.